server/leafnode.go

```diff
@@ -1362,8 +1362,7 @@ func (c *client) processLeafNodeConnect(s *Server, arg []byte, lang string) erro
 
 	// If we have JS enabled and the other side does as well we need to add in an import deny clause.
 	if jsConfigured && proto.JetStream {
-		// We should never have existing perms here, if that changes this needs to be reworked.
-		c.setPermissions(&Permissions{Publish: &SubjectPermission{Deny: []string{jsAllAPI}}})
+		c.mergePubDenyPermissions([]string{jsAllAPI})
```

Comment:
Based on the removed comment above, this is should not happen, so either comment was correct at one point of time but no longer the case, was never correct, or is still is and the merge approach is wrong. I have no idea which is which. I have ran tests and they all pass, so I would think the change is ok (and fixes the user report).

Reply:
Understood.